### PR TITLE
feat: airdrop transfer contract #ntrn-116 (recreated pr)

### PR DESCRIPTION
**Pull request by NeverHappened**
https://hadronlabs.atlassian.net/browse/NTRN-116

How to run integration tests:
1. Checkout gaia `v13.0.0` for building image, checkout neutron `main` branch.
2. in root: `make build`
3. `cd integration-tests`
4. `make build-all`
5. run `yarn`
6. run `yarn test`
7. tests should pass!

### CODE REVIEW

File: src/state.rs [hunk 1]
```diff
@@ -1,11 +1,49 @@
...
+}
+
+#[cw_serde]
+pub enum IbcCallbackState {
+    Response(RequestPacket, u64),      // request_packet, block_height
+    Timeout(RequestPacket, u64),       // request_packet, block_height
+    Error(RequestPacket, String, u64), // error with request_packet, details, block_height
+
+    OpenAckError(String, u64, String, String, String, String), // parse_error, block_height, port_id, channel_id, counterparty_channel_id, counterparty_version
 }
```
Comment (sotnikov-s) [inline src/state.rs:49]: afaik the structure of `IBC_CALLBACK_STATES` will be as array of arrays of 2-6 elements. given the fact there is 2 elements for both `Response` and `Timeout`, we won't be able to distinguish between them when reading query response.

like that. what is what here?
```
[
    [{packet}, 64],
    [{packet}, 66]
]
```

please prove me wrong or replace the enum variants with structures

**Review by sotnikov-s — COMMENTED** [review]
the rest LGTM so far

File: src/state.rs (hunk 1 above)
Comment (NeverHappened) [reply src/state.rs:49]: If I understand you correctly, no. Yo'll be able to differentiate between timeout or response. Structure is like this:
```
[
    Response(packet, u64],
    Timeout(packet, u64],
]
```

(example of check in tests https://github.com/neutron-org/neutron-airdrop-transfer/pull/3/files#diff-4b10913f4c3e7858207826d4e2e777308bebc273b75b7e6443880d8181111fb8R146)

Comment (sotnikov-s) [reply src/state.rs:49]: okay, agree. seems it's not as I described by me, but as
```
[
    {
        response: [{packet}, height]
    },
    {
        timeout: [{packet}, height]
    }
]
```

**Review by quasisamurai — APPROVED** [review]
testedACK

**Review by pr0n00gler — APPROVED** [review]

**Review by sotnikov-s — APPROVED** [review]

**Review by pr0n00gler — APPROVED** [review]

**Pull request merged**